src/backend/sycl_backend.cc

```diff
@@ -207,16 +253,18 @@ async_event sycl_backend::enqueue_host_task(size_t host_lane, const host_task_la
 async_event sycl_backend::enqueue_device_kernel(const device_id device, const size_t lane, const device_kernel_launcher& launch,
     std::vector<closure_hydrator::accessor_info> accessor_infos, const box<3>& execution_range, const std::vector<void*>& reduction_ptrs) //
 {
-	auto& queue = m_impl->get_device_queue(device, lane);
-	CELERITY_DETAIL_TRACY_ZONE_SCOPED("sycl::submit", Orange2);
-	auto event = queue.submit([&](sycl::handler& sycl_cgh) {
-		auto& hydrator = closure_hydrator::get_instance();
-		hydrator.arm(target::device, std::move(accessor_infos));
-		const auto launch_hydrated = hydrator.hydrate<target::device>(sycl_cgh, launch);
-		launch_hydrated(sycl_cgh, execution_range, reduction_ptrs);
+	return enqueue_device_work(device, [=, this] {
+		auto& queue = m_impl->get_device_queue(device, lane);
+		CELERITY_DETAIL_TRACY_ZONE_SCOPED("sycl::submit", Orange2);
+		auto event = queue.submit([&](sycl::handler& sycl_cgh) {
+			auto& hydrator = closure_hydrator::get_instance();
+			hydrator.arm(target::device, std::move(accessor_infos));
```

Comment:
This move is ineffective without also moving accessor_infos into the lambda capture (and making the lambda mutable).